### PR TITLE
Replace "MoveToWaypoint" behavior with "Move to Named Pose" objective

### PR DESCRIPTION
**Pull request by sjahr**
Part of https://github.com/PickNikRobotics/moveit_studio/pull/6571/files

# TODO
- [x] Test if everything is still working

### CODE REVIEW

File: src/picknik_ur_mock_hw_config/objectives/3_waypoint_pick_and_place.xml [hunk 1]
```diff
@@ -5,28 +5,28 @@
         <Control ID="Sequence">
             <!-- Clear snapshot, move to center pose, and open gripper -->
             <Action ID="ClearSnapshot" />
-            <Action ID="MoveToWaypoint" waypoint_name="Look at Machine" planning_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
+            <SubTree ID="Move to Named Pose" waypoint_name="Look at Table" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
```
Comment (Abishalini) [inline src/picknik_ur_mock_hw_config/objectives/3_waypoint_pick_and_place.xml:8]:
```suggestion
            <SubTree ID="Move to Named Pose" waypoint_name="Look at Machine" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
```

Look at Table waypoint does not exist for mock hw

File: src/picknik_ur_mock_hw_config/objectives/3_waypoint_pick_and_place.xml [hunk 2]
```diff
@@ -5,28 +5,28 @@
...
             <Action ID="MoveGripperAction" gripper_command_action_name="/robotiq_gripper_controller/gripper_cmd" position="0"/>
             <!-- Keep executing the pick and place sequence until failure -->
             <Decorator ID="KeepRunningUntilFailure">
                 <Control ID="Sequence">
-                    <!-- Pick object from left, put it down at the CNC machine, and go back to center pose -->
+                    <!-- Pick object from "Grab", put it down at "Place", and go back to center pose -->
                     <Control ID="Sequence">
-                        <Action ID="MoveToWaypoint" waypoint_name="Grasp Left" planning_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
+                        <SubTree ID="Move to Named Pose" waypoint_name="Pick Block" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
```
Comment (Abishalini) [inline src/picknik_ur_mock_hw_config/objectives/3_waypoint_pick_and_place.xml:15]:
```suggestion
                        <SubTree ID="Move to Named Pose" waypoint_name="Grasp Left" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
```

Looks like all the waypoints in this objective were copied over from gazebo config.

File: src/picknik_ur_gazebo_config/objectives/3_waypoint_pick_and_place.xml [hunk 3]
```diff
@@ -1,32 +1,32 @@
...
             <!-- Clear snapshot, move to center pose, and open gripper -->
             <Action ID="ClearSnapshot" />
-            <Action ID="MoveToWaypoint" waypoint_name="Look at Table" planning_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
+            <SubTree ID="Move to Named Pose" waypoint_name="Look at Table" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
             <Action ID="MoveGripperAction" gripper_command_action_name="/robotiq_gripper_controller/gripper_cmd" position="0"/>
             <!-- Keep executing the pick and place sequence until failure -->
             <Decorator ID="KeepRunningUntilFailure">
                 <Control ID="Sequence">
-                    <!-- Pick object from "Pick", put it down at "Place", and go back to center pose -->
+                    <!-- Pick object from "Grab", put it down at "Place", and go back to center pose -->
```
Comment (sea-bass) [inline src/picknik_ur_gazebo_config/objectives/3_waypoint_pick_and_place.xml:13]:
```suggestion
                    <!-- Pick object from "Pick", put it down at "Place", and go back to center pose -->

File: src/picknik_ur_gazebo_config/objectives/3_waypoint_pick_and_place.xml [hunk 4]
```diff
@@ -1,32 +1,32 @@
...
+                        <SubTree ID="Move to Named Pose" waypoint_name="Place" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
                         <Action ID="MoveGripperAction" gripper_command_action_name="/robotiq_gripper_controller/gripper_cmd" position="0"/>
-                        <Action ID="MoveToWaypoint" waypoint_name="Look at Table" planning_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
+                        <SubTree ID="Move to Named Pose" waypoint_name="Look at Table" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
                     </Control>
-                    <!-- Pick object from "Place", put it down at "Pick", and go back to center pose -->
+                    <!-- Pick object from "Place", put it down at "Grab", and go back to center pose -->
```
Comment (sea-bass) [inline src/picknik_ur_gazebo_config/objectives/3_waypoint_pick_and_place.xml:22]:
```suggestion
                    <!-- Pick object from "Place", put it down at "Pick", and go back to center pose -->

File: src/picknik_ur_mock_hw_config/objectives/3_waypoint_pick_and_place.xml [hunk 5]
```diff
@@ -5,28 +5,28 @@
...
             <!-- Clear snapshot, move to center pose, and open gripper -->
             <Action ID="ClearSnapshot" />
-            <Action ID="MoveToWaypoint" waypoint_name="Look at Machine" planning_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
+            <SubTree ID="Move to Named Pose" waypoint_name="Look at Machine" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
             <Action ID="MoveGripperAction" gripper_command_action_name="/robotiq_gripper_controller/gripper_cmd" position="0"/>
             <!-- Keep executing the pick and place sequence until failure -->
             <Decorator ID="KeepRunningUntilFailure">
                 <Control ID="Sequence">
-                    <!-- Pick object from left, put it down at the CNC machine, and go back to center pose -->
+                    <!-- Pick object from "Grab", put it down at "Place", and go back to center pose -->
```
Comment (sea-bass) [inline src/picknik_ur_mock_hw_config/objectives/3_waypoint_pick_and_place.xml:13]:
```suggestion
                    <!-- Pick object from left, put it down at the CNC machine, and go back to center pose -->

File: src/picknik_ur_mock_hw_config/objectives/3_waypoint_pick_and_place.xml [hunk 6]
```diff
@@ -5,28 +5,28 @@
...
+                        <SubTree ID="Move to Named Pose" waypoint_name="Place Block" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
                         <Action ID="MoveGripperAction" gripper_command_action_name="/robotiq_gripper_controller/gripper_cmd" position="0"/>
-                        <Action ID="MoveToWaypoint" waypoint_name="Look at Machine" planning_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
+                        <SubTree ID="Move to Named Pose" waypoint_name="Look at Table" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
                     </Control>
-                    <!-- Pick object from the CNC machine, put it down on the right, and go back to center pose -->
+                    <!-- Pick object from "Place", put it down at "Grab", and go back to center pose -->
```
Comment (sea-bass) [inline src/picknik_ur_mock_hw_config/objectives/3_waypoint_pick_and_place.xml:22]:
```suggestion
                    <!-- Pick object from the CNC machine, put it down on the right, and go back to center pose -->

File: src/picknik_ur_mock_hw_config/objectives/3_waypoint_pick_and_place.xml [hunk 7]
```diff
@@ -5,28 +5,28 @@
...
                         <Action ID="MoveGripperAction" gripper_command_action_name="/robotiq_gripper_controller/gripper_cmd" position="0.7929"/>
-                        <Action ID="MoveToWaypoint" waypoint_name="Look at Machine" planning_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
-                        <Action ID="MoveToWaypoint" waypoint_name="Grasp Machine" planning_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
+                        <SubTree ID="Move to Named Pose" waypoint_name="Look at Table" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
+                        <SubTree ID="Move to Named Pose" waypoint_name="Place Block" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
```
Comment (sea-bass) [inline src/picknik_ur_mock_hw_config/objectives/3_waypoint_pick_and_place.xml:18]:
```suggestion
                        <SubTree ID="Move to Named Pose" waypoint_name="Look at Machine" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
                        <SubTree ID="Move to Named Pose" waypoint_name="Grasp Machine" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>

File: src/picknik_ur_mock_hw_config/objectives/3_waypoint_pick_and_place.xml [hunk 8]
```diff
@@ -5,28 +5,28 @@
...
                         <Action ID="MoveGripperAction" gripper_command_action_name="/robotiq_gripper_controller/gripper_cmd" position="0"/>
-                        <Action ID="MoveToWaypoint" waypoint_name="Look at Machine" planning_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
+                        <SubTree ID="Move to Named Pose" waypoint_name="Look at Table" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
```
Comment (sea-bass) [inline src/picknik_ur_mock_hw_config/objectives/3_waypoint_pick_and_place.xml:20]:
```suggestion
                        <SubTree ID="Move to Named Pose" waypoint_name="Look at Machine" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>

File: src/picknik_ur_mock_hw_config/objectives/3_waypoint_pick_and_place.xml [hunk 9]
```diff
@@ -5,28 +5,28 @@
...
                     </Control>
-                    <!-- Pick object from the CNC machine, put it down on the right, and go back to center pose -->
+                    <!-- Pick object from "Place", put it down at "Grab", and go back to center pose -->
                     <Control ID="Sequence">
-                        <Action ID="MoveToWaypoint" waypoint_name="Grasp Machine" planning_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
+                        <SubTree ID="Move to Named Pose" waypoint_name="Place Block" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
```
Comment (sea-bass) [inline src/picknik_ur_mock_hw_config/objectives/3_waypoint_pick_and_place.xml:24]:
```suggestion
                        <SubTree ID="Move to Named Pose" waypoint_name="Grasp Machine" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>

File: src/picknik_ur_mock_hw_config/objectives/3_waypoint_pick_and_place.xml [hunk 10]
```diff
@@ -5,28 +5,28 @@
...
                         <Action ID="MoveGripperAction" gripper_command_action_name="/robotiq_gripper_controller/gripper_cmd" position="0.7929"/>
-                        <Action ID="MoveToWaypoint" waypoint_name="Look at Machine" planning_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
-                        <Action ID="MoveToWaypoint" waypoint_name="Grasp Left" planning_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
+                        <SubTree ID="Move to Named Pose" waypoint_name="Look at Table" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
+                        <SubTree ID="Move to Named Pose" waypoint_name="Pick Block" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
```
Comment (sea-bass) [inline src/picknik_ur_mock_hw_config/objectives/3_waypoint_pick_and_place.xml:27]:
```suggestion
                        <SubTree ID="Move to Named Pose" waypoint_name="Look at Machine" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
                        <SubTree ID="Move to Named Pose" waypoint_name="Grasp Left" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>

File: src/picknik_ur_mock_hw_config/objectives/3_waypoint_pick_and_place.xml [hunk 11]
```diff
@@ -5,28 +5,28 @@
...
                         <Action ID="MoveGripperAction" gripper_command_action_name="/robotiq_gripper_controller/gripper_cmd" position="0"/>
-                        <Action ID="MoveToWaypoint" waypoint_name="Look at Machine" planning_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
+                        <SubTree ID="Move to Named Pose" waypoint_name="Look at Table" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>
```
Comment (sea-bass) [inline src/picknik_ur_mock_hw_config/objectives/3_waypoint_pick_and_place.xml:29]:
```suggestion
                        <SubTree ID="Move to Named Pose" waypoint_name="Look at Machine" joint_group_name="manipulator" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" use_all_planners="false"/>

**Review by sea-bass — COMMENTED** [review]
found a couple more

**Comment by Abishalini** [comment]
Would it make sense to rename `Move To Named Pose` objective to `Move To Waypoint`?

**Comment by sjahr** [comment]
> Would it make sense to rename `Move To Named Pose` objective to `Move To Waypoint`?

Good question. I'd be more in favor of pose because, the named pose could also be a joint state defined in the srdf

**Comment by Abishalini** [comment]
@sjahr We can make that change as part of this issue - https://github.com/PickNikRobotics/moveit_studio/issues/6647

**Review by Abishalini — APPROVED** [review]

**Pull request merged**